src/colvarbias_meta.cpp

```diff
@@ -1782,62 +1779,27 @@ int colvarbias_meta::write_replica_state_file()
   if (cvm::debug()) {
     cvm::log("Writing replica state file for bias \""+name+"\"\n");
   }
-  // write down also the restart for the other replicas
-  cvm::backup_file(replica_state_file.c_str());
-  std::ostream *rep_state_os = cvm::proxy->output_stream(replica_state_file);
+
+  std::string const tmp_state_file(replica_state_file+".tmp");
+  std::ostream *rep_state_os = cvm::proxy->output_stream(tmp_state_file);
   if (rep_state_os == NULL) {
     cvm::error("Error: in opening file \""+
-               replica_state_file+"\" for writing.\n", FILE_ERROR);
+               tmp_state_file+"\" for writing.\n", FILE_ERROR);
     return FILE_ERROR;
   }
 
   rep_state_os->setf(std::ios::scientific, std::ios::floatfield);
 
   if (!write_state(*rep_state_os)) {
     cvm::error("Error: in writing to file \""+
-               replica_state_file+"\".\n", FILE_ERROR);
-    cvm::proxy->close_output_stream(replica_state_file);
+               tmp_state_file+"\".\n", FILE_ERROR);
+    cvm::proxy->close_output_stream(tmp_state_file);
     return FILE_ERROR;
   }
 
-  cvm::proxy->close_output_stream(replica_state_file);
-
-  // rep_state_os.setf(std::ios::scientific, std::ios::floatfield);
-  // rep_state_os << "\n"
-  //              << "metadynamics {\n"
-  //              << "  configuration {\n"
-  //              << "    name " << this->name << "\n"
-  //              << "    step " << cvm::step_absolute() << "\n";
-  // if (this->comm != single_replica) {
-  //   rep_state_os << "    replicaID " << this->replica_id << "\n";
-  // }
-  // rep_state_os << "  }\n\n";
-  // rep_state_os << "  hills_energy\n";
-  // rep_state_os << std::setprecision(cvm::cv_prec)
-  //              << std::setw(cvm::cv_width);
-  // hills_energy->write_restart(rep_state_os);
-  // rep_state_os << "  hills_energy_gradients\n";
-  // rep_state_os << std::setprecision(cvm::cv_prec)
-  //              << std::setw(cvm::cv_width);
-  // hills_energy_gradients->write_restart(rep_state_os);
-
-  // if ( (!use_grids) || keep_hills ) {
-  //   // write all hills currently in memory
-  //   for (std::list<hill>::const_iterator h = this->hills.begin();
-  //        h != this->hills.end();
-  //        h++) {
-  //     rep_state_os << *h;
-  //   }
-  // } else {
-  //   // write just those that are near the grid boundaries
-  //   for (std::list<hill>::const_iterator h = this->hills_off_grid.begin();
-  //        h != this->hills_off_grid.end();
-  //        h++) {
-  //     rep_state_os << *h;
-  //   }
-  // }
-  // rep_state_os << "}\n\n";
-  // rep_state_os.close();
+  cvm::proxy->close_output_stream(tmp_state_file);
+  cvm::backup_file(replica_state_file.c_str());
```

Comment:
Calling backup_file is unnecessary, as the file is not actually deleted while any client has it open. This also has the problem of creating a point in time when there is no file of the desired name. If you do want the preserved the behavior of having a backup file it would be better to use std::filesystem::create_hard_link().

Reply:
Sounds good, it is indeed not very useful with the renaming.  Deleted this call in b52827a